WordPressAuthenticator/GoogleSignIn/URL+GoogleSignIn.swift

```diff
@@ -0,0 +1,32 @@
+import Foundation
+
+extension URL {
+
+    // TODO: This is incomplete
+    static func googleSignInAuthURL(clientId: String) throws -> URL {
+        let baseURL = "https://accounts.google.com/o/oauth2/v2/auth"
+
+        let queryItems = [
+            ("client_id", clientId),
+            ("redirect_uri", redirectURI(from: clientId)),
+            ("response_type", "code")
+        ].map { URLQueryItem(name: $0.0, value: $0.1) }
+
+        if #available(iOS 16.0, *) {
+            return URL(string: baseURL)!.appending(queryItems: queryItems)
+        } else {
+            var components = URLComponents(string: baseURL)!
+            components.queryItems = queryItems
+            return try components.asURL()
+        }
+    }
+
+    private static func redirectURI(from clientId: String) -> String {
+        // Google's client id is in the form: 123-abc245def.apps.googleusercontent.com
+        // The redirect URI uses the reverse-DNS notation.
+        let reverseDNSClientId = clientId.split(separator: ".").reversed().joined(separator: ".")
+        // After that, we add "oautha2callback", as per GIDSignIn.m line 421 at
```

Comment:
Good question. I wasn't able to get it to work with anything else when prototyping #740 but I learned more since... 🤔 

I might give this another go.

To answer "Is it intended to use the same URL scheme as GoogleSignIn?" specifically, the [docs](https://developers.google.com/identity/protocols/oauth2/native-app#request-parameter-redirect_uri) list that as a valid value to use, and that's what their SDK does.

Thinking about it, "that's what their SDK does" is kinda the obvious path _for their SDK_, but it might be best for us to use a custom value, as you suggest. There's nothing in the docs that prevents it, so it might just be a matter of me figuring out where my early mistakes were.